x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form.tsx

```diff
@@ -39,28 +38,55 @@ export function SloEditForm({ slo }: Props) {
     http: { basePath },
     notifications: { toasts },
   } = useKibana().services;
+  const isEditMode = slo !== undefined;
 
   const methods = useForm({
     defaultValues: SLO_EDIT_FORM_DEFAULT_VALUES,
     values: transformSloResponseToCreateSloInput(slo),
     mode: 'all',
   });
-  const { watch, getFieldState, getValues, formState } = methods;
+  const { watch, getFieldState, getValues, formState, trigger } = methods;
 
-  const { isIndicatorSectionValid, isDescriptionSectionValid, isObjectiveSectionValid } =
+  const { isIndicatorSectionValid, isObjectiveSectionValid, isDescriptionSectionValid } =
     useSectionFormValidation({
       getFieldState,
       getValues,
       formState,
       watch,
     });
 
+  const [showObjectiveSection, setShowObjectiveSection] = useState<boolean>(isEditMode);
+  const [showDescriptionSection, setShowDescriptionSection] = useState<boolean>(isEditMode);
+  useEffect(() => {
+    if (!formState.isValidating && !showObjectiveSection && isIndicatorSectionValid) {
+      setShowObjectiveSection(true);
+    }
+
+    if (
+      !formState.isValidating &&
+      !showDescriptionSection &&
+      isIndicatorSectionValid &&
+      isObjectiveSectionValid
+    ) {
+      setShowDescriptionSection(true);
+    }
+  }, [
+    showObjectiveSection,
+    showDescriptionSection,
+    isIndicatorSectionValid,
+    isObjectiveSectionValid,
+    formState,
+  ]);
```

Comment:
Suggestion, but not necessary: break this out into a separate hook.

Reply:
Good idea, it grew a bit in complexity I agree.

Reply:
Done in https://github.com/elastic/kibana/pull/153790/commits/b975be3ec6c71a957a717efbb10cf4c054554574